netkan/netkan/webhooks/__init__.py

```diff
@@ -15,35 +15,34 @@
 from .github_mirror import github_mirror
 
 
-def create_app() -> Flask:
-    # Set up Discord logger so we can see errors
-    if setup_log_handler():
-        sys.excepthook = catch_all
-
-    app = Flask(__name__)
-
-    init_ssh(os.environ.get('SSH_KEY', ''), Path(Path.home(), '.ssh'))
-
-    # Set up config
-    app.config['secret'] = os.environ.get('XKAN_GHSECRET')
-    app.config['nk_repo'] = NetkanRepo(
-        init_repo(os.environ.get('NETKAN_REMOTE', ''), '/tmp/NetKAN', False))
-    app.config['ckm_repo'] = CkanMetaRepo(
-        init_repo(os.environ.get('CKANMETA_REMOTE', ''), '/tmp/CKAN-meta', False))
-    app.config['repos'] = [app.config['nk_repo'].git_repo, app.config['ckm_repo'].git_repo]
-    app.config['client'] = boto3.client('sqs')
-    sqs = boto3.resource('sqs')
-    app.config['inflation_queue'] = sqs.get_queue_by_name(
-        QueueName=os.environ.get('INFLATION_SQS_QUEUE'))
-    app.config['add_queue'] = sqs.get_queue_by_name(QueueName=os.environ.get('ADD_SQS_QUEUE'))
-    app.config['mirror_queue'] = sqs.get_queue_by_name(QueueName=os.environ.get('MIRROR_SQS_QUEUE'))
-
-    # Add the hook handlers
-    app.register_blueprint(errors)
-    app.register_blueprint(inflate)
-    app.register_blueprint(spacedock_inflate, url_prefix='/sd')
-    app.register_blueprint(spacedock_add, url_prefix='/sd')
-    app.register_blueprint(github_inflate, url_prefix='/gh')
-    app.register_blueprint(github_mirror, url_prefix='/gh')
-
-    return app
+class NetkanWebhooks(Flask):
+
+    def __init__(self, ssh_key: str) -> None:
+        super().__init__(__name__)
+
+        # Set up Discord logger so we can see errors
+        if setup_log_handler():
+            sys.excepthook = catch_all
+
+        init_ssh(ssh_key, Path(Path.home(), '.ssh'))
+
+        # Add the hook handlers
+        self.register_blueprint(errors)
+        self.register_blueprint(inflate)
+        self.register_blueprint(spacedock_inflate, url_prefix='/sd')
+        self.register_blueprint(spacedock_add, url_prefix='/sd')
+        self.register_blueprint(github_inflate, url_prefix='/gh')
+        self.register_blueprint(github_mirror, url_prefix='/gh')
+
+
+def create_app() -> NetkanWebhooks:
+    # Set config values for other modules to retrieve
+    current_config.setup(
+        os.environ.get('XKAN_GHSECRET', ''),
+        os.environ.get('NETKAN_REMOTE', ''), '/tmp/NetKAN',
+        os.environ.get('CKANMETA_REMOTE', ''), '/tmp/CKAN-meta',
+        os.environ.get('INFLATION_SQS_QUEUE', ''),
+        os.environ.get('ADD_SQS_QUEUE', ''),
+        os.environ.get('MIRROR_SQS_QUEUE', '')
```

Comment:
My thought was that if the config object itself just accepts these things as parameters rather than retrieving them itself, then it's more test-friendly. We could set up a `WebhooksConfig` object with whatever settings we like without having to worry about environment variables or whatever else.

Reply:
That is true, though we could use mock and replace the __init__ testing. But I'm very much ok with this as is. We should probably write the tests first and refactor later if that's what we want to do.